### PR TITLE
Add optional 'connect_args' to Snowflake connection method

### PR DESCRIPTION
**Pull request by malthe**
This allows for example the specifying of a private key.

See example on https://github.com/snowflakedb/snowflake-sqlalchemy.

### CODE REVIEW

File: ibis/backends/snowflake/__init__.py [hunk 1]
```diff
@@ -145,7 +145,13 @@ def version(self) -> str:
...
 
     def do_connect(
-        self, user: str, password: str, account: str, database: str, **kwargs: Any
+        self,
+        user: str,
+        password: str,
+        account: str,
+        database: str,
+        connect_args: Dict[str, Any] = None,
+        **kwargs: Any,
```
[the remark this answers is not in the document]
Comment (malthe) [reply ibis/backends/snowflake/__init__.py:154]: The difference is that `kwargs` go to the URL where `connect_args` go to `create_engine`.

But there's a problem with `password` as well in my use-case, I need to provide an empty string for it to work. Makes sense when you look at the function signature. It's a minor detail all things considered.

File: ibis/backends/snowflake/__init__.py [hunk 2]
```diff
@@ -145,7 +145,13 @@ def version(self) -> str:
             return con.exec_driver_sql("SELECT CURRENT_VERSION()").scalar()
 
     def do_connect(
-        self, user: str, password: str, account: str, database: str, **kwargs: Any
+        self,
+        user: str,
+        password: str,
+        account: str,
+        database: str,
+        connect_args: dict[str, Any] = None,
```
Comment (cpcloud) [inline ibis/backends/snowflake/__init__.py:153]:
```suggestion
        connect_args: Mapping[str, Any] | None = None,
```

Comment (malthe) [reply ibis/backends/snowflake/__init__.py]: Fixed in 22b57af01f14ba243b9d3d16565fe954fbff24c7. Thanks!

**Review by cpcloud — APPROVED** [review]
LGTM, I'll run the tests against the snowflake backend!

**Comment by cpcloud** [comment]
Snowflake backend tests look good:

```
cloud in 🌐 albatross in …/ibis on  snowflake-connect-args is 📦 v4.1.0 via 🐍 v3.10.10 via ❄️  impure (ibis-3.10.10-env) took 1m17s
❯ pytest -m snowflake --snapshot-update -n auto --dist loadgroup -q
bringing up nodes...
......x.............................x.......................................................xx...x....x.x....x....................xx..................x.....x.x.x....................................x.......................x.....x.x [ 26%]
...x.x.x........x......x......x............................x..........x...x..............................x...x.x.........xx.xxxx.....x.xx.xxx.x.x......x.......x.......x......................x........x.xx.....xx...x..x...x......... [ 52%]
x...xx....xx..xx.s...x........................x....x...........x...x......x....x.....xxx............x..xxx.x..xx..........x.xxx.x...x.x........x.............x.xx....xx.......xxx..........x.x......x.x.x..x.x.x..x..x...x.x.x..x....x [ 78%]
....x...xxx......x...x..x.x..............x..xx.xx.....x...x.x....xx.......xs..........xxx...x..s.....xs...........x........................xx.x...xs..s.sx....xxs.x.s..xx....xx.x.sx.........x                                         [100%]
719 passed, 10 skipped, 151 xfailed in 124.20s (0:02:04)
```

**Pull request merged**